### PR TITLE
Add typescript informations in coding standards

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | We need to specify some information about typescript in coding standards

### CODE REVIEW

File: src/content/1.7/development/coding-standards/_index.md [hunk 1]
```diff
@@ -169,6 +169,14 @@ You can fix auto-fixable errors using this command:
 npm run scss-fix
 ```
 
+## TypeScript
+Starting on {{< minver v="1.7.9" >}}, most of `admin-dev/themes/new-theme/js` files have been migrated to TypeScript. This provide a better maintainability by using types on most of our classes and functions.
```
Comment (eternoendless) [inline src/content/1.7/development/coding-standards/_index.md:173]: This doesn't belong to the 1.7 documentation ;)

**Comment by matks** [comment]
Hello,

I have modified your PR to target the `1.7.x` branch.

We have modified the devdocs inner working in order to welcome multiple documentation versions. This is because soon we will have a 1.7 documentation and a 8.0 documentation, matching PrestaShop 1.7 or PrestaShop 8.0 softwares.

Read more on https://devdocs.prestashop.com/1.7/contribute/documentation/how/documentation-versions/

Please also note that now that we have multiple versions inside one documentation, we had to build a workflow that can handle this setup. This repository https://github.com/PrestaShop/docs now only contains the content of the documentation, and https://github.com/PrestaShop/devdocs-site contains the sources and the deployment workflow.

Because of this, some files have moved and you might have to rebase your Pull Request to match the new structure of the repository.

File: development/coding-standards/_index.md [hunk 2]
```diff
@@ -171,6 +171,14 @@ You can fix auto-fixable errors using this command:
 npm run scss-fix
 ```
 
+## TypeScript
+Most of `admin-dev/themes/new-theme/js` files are using TypeScript. This provide a better maintainability by using types on most of our classes and functions.
```
Comment (eternoendless) [inline development/coding-standards/_index.md]:
```suggestion

Most of `admin-dev/themes/new-theme/js` files are coded in TypeScript. Classes and functions in .ts files must be strictly typed. 
```

File: development/coding-standards/_index.md [hunk 3]
```diff
@@ -171,6 +171,14 @@ You can fix auto-fixable errors using this command:
 npm run scss-fix
 ```
 
+## TypeScript
```
Comment (eternoendless) [inline development/coding-standards/_index.md]: Can you please move this block under the Javascript section? This is more like a third-level block.

File: development/coding-standards/_index.md [hunk 4]
```diff
@@ -171,6 +171,14 @@ You can fix auto-fixable errors using this command:
...
+Most of `admin-dev/themes/new-theme/js` files are using TypeScript. This provide a better maintainability by using types on most of our classes and functions. 
+
+You are able to get global types in the `admin-dev/themes/new-theme/js/types` folder and some types library are imported from npm using the `@types` namespace and automatically imported by TypeScript.
+
+The `tsconfig.json` and `webpack` configuration files are adding a `@PSTypes` relative path to the `admin-dev/themes/new-theme/js/types` folder, because `@types` is a reserved folder.
```
Comment (eternoendless) [inline development/coding-standards/_index.md]: I'm struggling to understand this part. It doesn't look like coding standards to me.

Comment (NeOMakinG) [reply development/coding-standards/_index.md]: We should tell peoples that our types are in the @PSTypes relative path, and external types are in the @types namespace, but @types is a standard so we are maybe not forced to explain it as it's TS standard

**Review by matks — COMMENTED** [review]
Feels OK to me but @eternoendless review is needed IMO

**Review by PierreRambaud — APPROVED** [review]

File: development/coding-standards/_index.md [hunk 5]
```diff
@@ -147,6 +147,12 @@ Javascript files MUST follow the [Airbnb Javascript style guide](https://github.
 
 You can run the linter to help you comply with these coding standards:
 
+## TypeScript
+
+Most of `admin-dev/themes/new-theme/js` files are coded in TypeScript. Classes and functions in .ts files must be strictly typed.
```
Comment (Progi1984) [inline development/coding-standards/_index.md]:
```suggestion
All `admin-dev/themes/new-theme/js` files are coded in TypeScript. Classes and functions in .ts files must be strictly typed. 
```

**Review by Progi1984 — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @NeOMakinG

**Pull request merged**